docs/core/docker/publish-as-container.md

```diff
@@ -0,0 +1,349 @@
+
+```dotnetcli
+dotnet run
+Building...
+info: DotNet.ContainerImage.Worker[0]
+      Worker running at: 10/18/2022 08:56:00 -05:00
+info: Microsoft.Hosting.Lifetime[0]
+      Application started. Press Ctrl+C to shut down.
+info: Microsoft.Hosting.Lifetime[0]
+      Hosting environment: Development
+info: Microsoft.Hosting.Lifetime[0]
+      Content root path: .\Worker
+info: DotNet.ContainerImage.Worker[0]
+      Worker running at: 10/18/2022 08:56:01 -05:00
+info: DotNet.ContainerImage.Worker[0]
+      Worker running at: 10/18/2022 08:56:02 -05:00
+info: DotNet.ContainerImage.Worker[0]
+      Worker running at: 10/18/2022 08:56:03 -05:00
+info: Microsoft.Hosting.Lifetime[0]
+      Application is shutting down...
+Attempting to cancel the build...
+```
+
+The worker template loops indefinitely. Use the cancel command <kbd>Ctrl+C</kbd> to stop it.
+
+## Add NuGet package
+
+The .NET 7 SDK will eventually be capable of publishing .NET apps as containers without the use of the [Microsoft.NET.Build.Containers NuGet package](https://libraries.io/nuget/Microsoft.NET.Build.Containers). Until that time, this package is required. To add the `Microsoft.NET.Build.Containers` NuGet package to the worker template, run the following [dotnet add package](../tools/dotnet-add-package.md) command:
+
+```dotnetcli
+dotnet add package Microsoft.NET.Build.Containers
+```
+
+## Set the container image name
+
+There are various configuration options available when publishing an app as a container. For more information, see [Configure container image](#configure-container-image).
+
+By default, the container image name is the `AssemblyName` of the project. If that name is invalid as a container image name, you can override it by specifying a `ContainerImageName` as shown in the following:
+
+:::code language="xml" source="snippets/Worker/DotNet.ContainerImage.csproj" highlight="8":::
+
+## Publish .NET app
+
+To publish the .NET app as a container, use the following `dotnet publish` command:
+
+```dotnetcli
+dotnet publish --os linux --arch x64 /t:PublishContainer -c Release
+```
+
+The preceding .NET CLI command, publishes the app as a container:
+
+- Targeting Linux as the OS (`--os linux`).
+- Specifying an x64 architecture (`--arch x64`).
+- Using the release configuration (`-c Release`).
+
+> [!IMPORTANT]
+> To build the container locally, you must have the Docker daemon running. If it isn't running when you attempt to publish the app as a container, you'll experience an error similar to the following:
+>
+> ```console
+> ..\.nuget\packages\microsoft.net.build.containers\0.1.8\build\Microsoft.NET.Build.Containers.targets(66,9): error MSB4018: The "CreateNewImage" task failed unexpectedly. [..\Worker\DotNet.ContainerImage.csproj]
```

Comment:
By 7 GA we should have better errors here - I'll ping you on any PRs related to that, ok?

Reply:
Sounds good.